src/ifcgeom/IfcGeomIterator.h

```diff
@@ -330,6 +371,14 @@ namespace IfcGeom {
 									break;
 								}
 							}
+
+                            foreach(const boost::regex& r, names_to_include_or_exclude) {
+                                if (boost::regex_match((*jt)->Name(), r)) {
```

Comment:
Ah, fyi, when merging my own PR I found one small oversight. `Name` is an optional attribute so its existence to be checked with `hasName()` first. This should be urgently addressed on a schema level in 0.6 so that optional attributes are of type `boost::optional<T>` (#50).